infor-design/enterprise-wc
# 1482 - Add min/max width/height settings to IdsTextArea

**Explain the details for making this change. What existing problem does the pull request solve?**
This PR enhances the settings that define height/width/size of IdsTextArea elements:
- Removes `autogrow-max-height` in favor of a more generic `max-height` setting
- Also adds `min-height`, `max-width`, and `min-width` settings
- Re-worked `resizable` setting to also allow horizontal resize, and resize in both directions simultaneously
- Re-worked `autogrow` setting to not shrink a previously, manually-resized IdsTextArea

**Related github/jira issue (required)**:
Closes #1482

**Steps necessary to review your pull request (required)**:
Pull/build/run, then test the following scenarios:

_New max/min settings:_
- Open http://localhost:4300/ids-textarea/sizes.html
- See that all examples on this page load initially as if they are normal-sized textarea field
- Try resizing all examples.  They should work in the directions they specify, and should have min/max limits as specified.  Anything labeled with a `min` size can be shrunk very small in that direction.  Anything labeled with a `max` size will not grow beyond that size.  Anything without a `max` size should grow indefinitely (except that these `max-width` fields are constrained within IdsLayoutGridCells and are bound to the maximum size of those grid cells).
- Try typing into the `autogrow` sample.  If you add a lot of line breaks and content, the textarea should grow as input stretches past its boundary.  It should not shrink after the size is modified.

_Test existing samples:_
- Open http://localhost:4300/ids-textarea/example.html
- See that all examples should look/behave as they did before (especially check the field sizes at the bottom, which all should be the same initial width they were before)
- Also smoke test other samples: http://localhost:4300/ids-textarea

**Included in this Pull Request**:
